modules/diskq/qdisk.c

```diff
@@ -567,6 +567,22 @@ _load_non_reliable_queues(QDisk *self, GQueue *qout, GQueue *qbacklog, GQueue *q
 
 #define _clear(obj) memset(&obj, 0, sizeof(obj));
 
+static gint64
+_number_of_messages(QDisk *self)
```

Comment:
This should be kept as an implementation details of each type of diskq. (virtual function)

Reply:
Just an informal comment before any virtual function is implemented (I still need to check the entire disk-buffer code): there is a `qdisk_get_length` function which only deals with `QDisk->hdr->length` member.
I would check that one before.

I just wonder if we have write-, read- and backlog_head why can't the `hdr->length` show the actual number of messages... This is a question to myself though...

Reply:
To use the already existing `LogQueueDisk->get_length` virtual function, we should bring this debug message back into `LogQueueDisk->load_queue`. Which would be far more superior than double implementing the get length logic :(

Unfortunately this change: https://github.com/syslog-ng/syslog-ng/commit/a6ed296b0b63fcd8a456c9f13c3a5e8aac07b4ce widens the scope of such a refactor. Should I move that fix too, or leave the current version?

My thinking: where should the fix be? In the `qdisk.c`, because it is file handling related. Or in the `logqueue-disk-non-reliable.c`, because it is related to handling non reliable diskq's.

Reply:
I do not recommend moving the `_reset_queue_pointers` function call from it's current location, because:

- We have to change the publicity of the `_reset_queue_pointers` function or the `QDiskFileHeader` struct.
- I would rather keep the details of header handling inside qdisk.c and do not spread this implementation detail: https://github.com/syslog-ng/syslog-ng/blob/master/modules/diskq/qdisk.c#L862